docs/reference/analysis/tokenfilters/pattern_replace-tokenfilter.asciidoc

```diff
@@ -4,23 +4,157 @@
+==== Example
+
+The following <<indices-analyze,analyze API>> request uses the `pattern_replace`
+filter to prepend `watch` to the substring `dog` in `foxes jump lazy dogs`.
+
+[source,console]
+----
+GET /_analyze
+{
+  "tokenizer": "whitespace",
+  "filter": [
+    {
+      "type": "pattern_replace",
+      "pattern": "(dog)",
+      "replacement": "watch$1"
+    }
+  ],
+  "text": "foxes jump lazy dogs"
+}
+----
+
+The filter produces the following tokens.
+
+[source,text]
+----
+[ foxes, jump, lazy, watchdogs ]
+----
+
+////
+[source,console-result]
+----
+{
+  "tokens": [
+    {
+      "token": "foxes",
+      "start_offset": 0,
+      "end_offset": 5,
+      "type": "word",
+      "position": 0
+    },
+    {
+      "token": "jump",
+      "start_offset": 6,
+      "end_offset": 10,
+      "type": "word",
+      "position": 1
+    },
+    {
+      "token": "lazy",
+      "start_offset": 11,
+      "end_offset": 15,
+      "type": "word",
+      "position": 2
+    },
+    {
+      "token": "watchdogs",
+      "start_offset": 16,
+      "end_offset": 20,
+      "type": "word",
+      "position": 3
+    }
+  ]
+}
+----
+////
+
+[[analysis-pattern-replace-tokenfilter-configure-parms]]
+==== Configurable parameters
+
+`all`::
+(Optional, boolean)
+If `true`, all substrings matching the `pattern` parameter's regular expression
+are replaced. If `false`, the filter replaces only the first matching substring
+in each token. Defaults to `true`.
+
+`pattern`::
+(Required, string)
+Regular expression, written in
+http://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html[Java's
+regular expression syntax]. The filter replaces token substrings matching this
+pattern with the substring in the `replacement` parameter.
+
+`replacement`::
+(Optional, string)
+Replacement substring. Defaults to an empty substring (`""`).
+
+[[analysis-pattern-replace-tokenfilter-customize]]
+==== Customize and add to an analyzer
 
-The pattern replace token filter uses
-http://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html[Java Regular Expressions].
+To customize the `pattern_replace` filter, duplicate it to create the basis
+for a new custom token filter. You can modify the filter using its configurable
+parameters.
 
-A badly written regular expression could run very slowly or even throw a
-StackOverflowError and cause the node it is running on to exit suddenly.
+The following <<indices-create-index,create index API>> request
+configures a new <<analysis-custom-analyzer,custom analyzer>> using a custom
+`pattern_replace` filter, `my_pattern_replace_filter`.
 
-Read more about http://www.regular-expressions.info/catastrophic.html[pathological regular expressions and how to avoid them].
+The `my_pattern_replace_filter` filter uses the regular expression `\\p{Sc}` to
+match and remove currency symbols, such as `£`, `€`, or `$`. The filter's `all`
+parameter is `false`, meaning only the first matching symbol in each token is
+removed.
```

Comment:
I did not know `\\p{Sc}` was a thing so thanks for educating me :) 
While these more esoteric functions work in token filters they are not supported in [RegExp](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html) query syntax.

I wonder if it's worth doing one of two things here:
A) Add a footnote to the effect of "* note this particular expression wouldn't work with RegExp queries which use a different regex parser" or
B) Use a different example that works in both "pattern_replace" token filters and RegExp queries.

Reply:
Thanks @markharwood. With 992ab1cd8e4, I updated the example to use a pattern that compatible with the RegExp query.